app/src/main/java/com/example/easycloset/UploadFragment.java

```diff
@@ -0,0 +1,161 @@
+        photoFile = getPhotoFileUri(photoFileName);
+        // wrap File object into a content provider
+        Uri fileProvider = FileProvider.getUriForFile(requireContext(), "com.codepath.fileprovider", photoFile);
+        intent.putExtra(MediaStore.EXTRA_OUTPUT, fileProvider);
+        // If you call startActivityForResult() using an intent that no app can handle, your app will crash.
+        // So as long as the result is not null, it's safe to use the intent.
+        if (intent.resolveActivity(requireContext().getPackageManager()) != null) {
+            // Start the image capture intent to take photo
+            startActivityForResult(intent, constant.getCAPTURE_IMAGE_ACTIVITY_REQUEST_CODE());
+        }
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        if (requestCode == constant.getCAPTURE_IMAGE_ACTIVITY_REQUEST_CODE()) {
+            if (resultCode == RESULT_OK) {
+                // by this point we have the camera photo on disk
+                Bitmap takenImage = BitmapFactory.decodeFile(photoFile.getAbsolutePath());
+                // RESIZE BITMAP, see section below
+                // Load the taken image into a preview
+                ivPicture.setImageBitmap(takenImage);
+            } else { // Result was a failure
+                Toast.makeText(getContext(), "Picture wasn't taken!", Toast.LENGTH_SHORT).show();
+            }
+        }
+    }
+
+    private File getPhotoFileUri(String photoFileName) {
+        // Get safe storage directory for photos
+        // Use `getExternalFilesDir` on Context to access package-specific directories.
+        // This way, we don't need to request external read/write runtime permissions.
+        String TAG = ".CameraActivity";
+        File mediaStorageDir = new File(requireContext().getExternalFilesDir(Environment.DIRECTORY_PICTURES), TAG);
+        // Create the storage directory if it does not exist
+        if (!mediaStorageDir.exists()) {
+            mediaStorageDir.mkdirs();
+        }
+        // Return the file target for the photo based on filename
+        return new File(mediaStorageDir.getPath() + File.separator + photoFileName);
+    }
+
+    @SuppressLint("NotifyDataSetChanged")
+    private void saveItem(String colour, String category, File photoFile) {
+        Item item = new Item();
+        item.setColour(colour);
+        item.setCategory(category);
+        item.setImage(new ParseFile(photoFile));
+        item.saveInBackground(e -> {
+            if (e != null) {
+                Toast.makeText(getContext(), "Error while saving", Toast.LENGTH_SHORT).show();
+            } else {
+                etColour.setText("");
+                ivPicture.setImageResource(0);
+                Toast.makeText(getContext(), "Saved", Toast.LENGTH_SHORT).show();
+                activity.getClosetFragment().getAllItems().add(0, item);
```

Comment:
direct access to a fragment inside of activity is not good approach, because in general fragments may be inside different activities , e.g. UploadFragment might be hosted not by MainActivity but somewhere else.
Instead of doing that, define interface in MainActivity, pass implementation to this fragment and call methods from interface

Reply:
fixed. created an interface and method in the mainActivity and call the method from the interface in the fragments